fabric-registry-sync-v0/src/main/java/net/fabricmc/fabric/mixin/registry/sync/SimpleRegistryMixin.java

```diff
@@ -186,7 +186,7 @@ public void setPre(int id, RegistryKey<T> registryId, T object, Lifecycle lifecy
 			RegistryEntry.Reference<T> oldObject = idToEntry.get(registryId.getValue());
 
 			if (oldObject != null && oldObject != object) {
```

Comment:
Isn't there still a bug here since `oldObject` (a `RegistryEntry.Reference<T>`) can never be identical to a `T`?